packages/netsuite-adapter/src/adapter_creator.ts

```diff
@@ -104,13 +131,32 @@ const netsuiteConfigFromConfig = (config: Readonly<InstanceElement> | undefined)
 }
 
 const netsuiteCredentialsFromCredentials = (credentials: Readonly<InstanceElement>): Credentials =>
-  credentials.value as Credentials
+  ({
+    accountId: credentials.value.accountId,
+    sdfTokenId: credentials.value.tokenId,
+    sdfTokenSecret: credentials.value.tokenSecret,
+    suiteAppTokenId: credentials.value.suiteAppTokenId === '' ? undefined : credentials.value.suiteAppTokenId,
+    suiteAppTokenSecret: credentials.value.suiteAppTokenSecret === '' ? undefined : credentials.value.suiteAppTokenSecret,
```

Comment:
nit: you can modify the code like defined below and then in rows 145 and 170 check only `credentials.suiteAppTokenId && credentials.suiteAppTokenSecret`
```suggestion
    suiteAppTokenId: credentials.value.suiteAppTokenId,
    suiteAppTokenSecret: credentials.value.suiteAppTokenSecret,
```

Reply:
I would prefer `suiteAppTokenId` and `suiteAppTokenSecret` to be undefined if not passed because, unlike `''`, it will make the compile force us to check if they are passed before using them. But regardless, I can still switch the conditions in 145 and 170 as you suggested.